wgsl/index.bs

```diff
@@ -9061,20 +9064,43 @@ See [[#function-calls]].
     [=Component-wise=] when |T| is a vector.
     (GLSLstd450Acos)
 
+  <tr algorithm="acosh">
+    <td>|T| is [FLOATING]
+    <td class="nowrap">`acosh(`|e|`:` |T| `) -> ` |T|
+    <td>Returns the hyperbolic arc cosine of |e|.
+    The result is undefined when |e| &lt; 1.<br>
```

Comment:
why is this OK for WGSL to have undefined results for these functions?

Reply:
I guess the other thing to say that these are the real-valued counterparts to complex functions.  
The arc-* blah functions are function inverses, and there are many valid inverses for any particular z input value.

You always need a convention to pick a single one of those valid answers; that's called picking a "branch". And the inverse function is discontinuous in certain regions (in this case a half-infinite line segment).   So there is no "good" answer to polyfill there that makes sense.

A little more here: https://www.johndcook.com/blog/2016/12/17/branch-cuts-and-common-lisp/ 
or more abstractly, here https://mathworld.wolfram.com/BranchCut.html

Reply:
Take 3:
I guess we can keep it simple and just look at the real-valued functions cosh, sinh, tanh.
The acosh, asinh, atanh functions are their respective functional inverses.
Staring at the graphs here: https://en.wikipedia.org/wiki/Hyperbolic_functions you can see that acosh(y) is not defined for  y < 1,   atanh(y) is not defined for |y| > 1.

Reply:
No need to explain *why* the functions are expecting particular range for arguments. That is math, and that is crystal clear.
My question was more about having the "undefined result" at all.
While I agree that it may not be super useful to define what they do out of the expected range, but same goes for other functions which we (so far) decided to define very concretely, like that integer division by -1.
I'd like to see some consistency or set of principles we'd follow for determining when undefined result is OK and when it isn't.

Reply:
This probably won't be a satisfying answer, but I think some of this boils down to where do we expect that users think there is a single well-defined answer? Integer division is one of those cases, but hyperbolic math functions are not. It's a judgement call. As such there may be some compromise involved.